### PR TITLE
Added check for GNUtls in FIPS140 mode

### PR DESCRIPTION
**Pull request by pablo-herranz**
Verify that FIPS bci container is in FIPS140 mode to be compliant with Vendor Affirmation tests

https://progress.opensuse.org/issues/167683

[CI:TOXENVS] fips

### CODE REVIEW

File: tests/test_fips.py [hunk 1]
```diff
@@ -195,6 +195,10 @@ def test_gnutls_binary(container_per_test: ContainerData) -> None:
         "mv fips-test-gnutls /bin/fips-test-gnutls"
     )
 
+    assert "NOT" not in container_per_test.connection.check_output(
```
Comment (grisu48) [inline tests/test_fips.py:198]: This check is wrong. The requirement states:

`should return "library is in FIPS140-3 mode"or "library is in FIPS140-2 mode" or "library is in FIPS140 mode"`

So just checking for NOT is not what the requirement states.

Comment (pablo-herranz) [reply tests/test_fips.py]: Changed :)

File: tests/test_fips.py [hunk 2]
```diff
@@ -4,6 +4,7 @@
 """
 
 from pathlib import Path
+from re import search as regexsearch
```
Comment (ricardobranco777) [inline tests/test_fips.py:7]: Why not use `re.search` which is more idiomatic?

File: tests/test_fips.py [hunk 3]
```diff
@@ -189,12 +190,19 @@ def test_gnutls_binary(container_per_test: ContainerData) -> None:
...
+    c = container_per_test.connection
+
+    c.check_output(
         "zypper --gpg-auto-import-keys -n ref && zypper -n install gcc gnutls gnutls-devel && zypper -n clean && "
         "gcc -Og -g3 fips-test-gnutls.c -Wall -Wextra -Wpedantic -lgnutls -o fips-test-gnutls && "
         "mv fips-test-gnutls /bin/fips-test-gnutls"
     )
 
+    assert regexsearch(
+        r"library is in FIPS140(-3|-2|) mode",
```
Comment (ricardobranco777) [inline tests/test_fips.py:202]:
```suggestion
        r"library is in FIPS140-(2|3) mode",
```

**Pull request closed without merging**